x/ccv/consumer/keeper/validators.go

```diff
@@ -60,31 +60,73 @@ func (k Keeper) ApplyCCValidatorChanges(ctx sdk.Context, changes []abci.Validato
 // IterateValidators - unimplemented on CCV keeper but perform a no-op in order to pass the slashing module InitGenesis.
 // It is allowed since the condition verifying validator public keys in HandleValidatorSignature (x/slashing/keeper/infractions.go) is removed
 // therefore it isn't required to store any validator public keys to the slashing states during genesis.
-func (k Keeper) IterateValidators(sdk.Context, func(index int64, validator stakingtypes.ValidatorI) (stop bool)) {
+func (k Keeper) IterateValidators(ctx sdk.Context, f func(index int64, validator stakingtypes.ValidatorI) (stop bool)) {
+	lastSovereignHeight := k.LastSovereignHeight(ctx)
+	if lastSovereignHeight == 0 || ctx.BlockHeight() <= lastSovereignHeight {
+		if k.stakingKeeper == nil {
+			panic("empty staking keeper")
+		}
+
+		k.stakingKeeper.IterateValidators(ctx, f)
+	}
 }
 
 // Validator - unimplemented on CCV keeper
 func (k Keeper) Validator(ctx sdk.Context, addr sdk.ValAddress) stakingtypes.ValidatorI {
+	lastSovereignHeight := k.LastSovereignHeight(ctx)
+	if lastSovereignHeight == 0 || ctx.BlockHeight() <= lastSovereignHeight {
+		if k.stakingKeeper == nil {
+			panic("empty staking keeper")
+		}
+
+		return k.stakingKeeper.Validator(ctx, addr)
+	}
 	panic("unimplemented on CCV keeper")
 }
 
 // IsJailed returns the outstanding slashing flag for the given validator adddress
 func (k Keeper) IsValidatorJailed(ctx sdk.Context, addr sdk.ConsAddress) bool {
+	lastSovereignHeight := k.LastSovereignHeight(ctx)
+	if lastSovereignHeight == 0 || ctx.BlockHeight() <= lastSovereignHeight {
+		if k.stakingKeeper == nil {
+			panic("empty staking keeper")
+		}
+
+		return k.stakingKeeper.IsValidatorJailed(ctx, addr)
+	}
 	return k.OutstandingDowntime(ctx, addr)
```

Comment:
When should control flow reach this return statement? Wont `lastSovereignHeight` be `0` if it's not set?

Reply:
In case block height is after last sovereign height as well as preCCV state is removed.

Reply:
LastSovereign height is set when genesis validator set is returned.